portal/src/util/deviceinfo.ts

```diff
@@ -0,0 +1,149 @@
+  "iPad4,6": "iPad Mini 2", // (2nd Generation iPad Mini)
+  "iPad4,7": "iPad Mini 3", // (3rd Generation iPad Mini)
+  "iPad4,8": "iPad Mini 3", // (3rd Generation iPad Mini)
+  "iPad4,9": "iPad Mini 3", // (3rd Generation iPad Mini)
+  "iPad5,1": "iPad Mini 4", // (4th Generation iPad Mini)
+  "iPad5,2": "iPad Mini 4", // (4th Generation iPad Mini)
+  "iPad5,3": "iPad Air 2", // 6th Generation iPad (iPad Air 2)
+  "iPad5,4": "iPad Air 2", // 6th Generation iPad (iPad Air 2)
+  "iPad6,3": "iPad Pro 9.7-inch", // iPad Pro 9.7-inch
+  "iPad6,4": "iPad Pro 9.7-inch", // iPad Pro 9.7-inch
+  "iPad6,7": "iPad Pro 12.9-inch", // iPad Pro 12.9-inch
+  "iPad6,8": "iPad Pro 12.9-inch", // iPad Pro 12.9-inch
+  "iPad6,11": "iPad (5th generation)", // Apple iPad 9.7 inch (5th generation) - WiFi
+  "iPad6,12": "iPad (5th generation)", // Apple iPad 9.7 inch (5th generation) - WiFi + cellular
+  "iPad7,1": "iPad Pro 12.9-inch", // 2nd Generation iPad Pro 12.5-inch - Wifi
+  "iPad7,2": "iPad Pro 12.9-inch", // 2nd Generation iPad Pro 12.5-inch - Cellular
+  "iPad7,3": "iPad Pro 10.5-inch", // iPad Pro 10.5-inch - Wifi
+  "iPad7,4": "iPad Pro 10.5-inch", // iPad Pro 10.5-inch - Cellular
+  "iPad7,5": "iPad (6th generation)", // iPad (6th generation) - Wifi
+  "iPad7,6": "iPad (6th generation)", // iPad (6th generation) - Cellular
+  "iPad7,11": "iPad (7th generation)", // iPad 10.2 inch (7th generation) - Wifi
+  "iPad7,12": "iPad (7th generation)", // iPad 10.2 inch (7th generation) - Wifi + cellular
+  "iPad8,1": "iPad Pro 11-inch (3rd generation)", // iPad Pro 11 inch (3rd generation) - Wifi
+  "iPad8,2": "iPad Pro 11-inch (3rd generation)", // iPad Pro 11 inch (3rd generation) - 1TB - Wifi
+  "iPad8,3": "iPad Pro 11-inch (3rd generation)", // iPad Pro 11 inch (3rd generation) - Wifi + cellular
+  "iPad8,4": "iPad Pro 11-inch (3rd generation)", // iPad Pro 11 inch (3rd generation) - 1TB - Wifi + cellular
+  "iPad8,5": "iPad Pro 12.9-inch (3rd generation)", // iPad Pro 12.9 inch (3rd generation) - Wifi
+  "iPad8,6": "iPad Pro 12.9-inch (3rd generation)", // iPad Pro 12.9 inch (3rd generation) - 1TB - Wifi
+  "iPad8,7": "iPad Pro 12.9-inch (3rd generation)", // iPad Pro 12.9 inch (3rd generation) - Wifi + cellular
+  "iPad8,8": "iPad Pro 12.9-inch (3rd generation)", // iPad Pro 12.9 inch (3rd generation) - 1TB - Wifi + cellular
+  "iPad11,1": "iPad Mini 5", // (5th Generation iPad Mini)
+  "iPad11,2": "iPad Mini 5", // (5th Generation iPad Mini)
+  "iPad11,3": "iPad Air (3rd generation)",
+  "iPad11,4": "iPad Air (3rd generation)",
+  "AppleTV2,1": "Apple TV", // Apple TV (2nd Generation)
+  "AppleTV3,1": "Apple TV", // Apple TV (3rd Generation)
+  "AppleTV3,2": "Apple TV", // Apple TV (3rd Generation - Rev A)
+  "AppleTV5,3": "Apple TV", // Apple TV (4th Generation)
+  "AppleTV6,2": "Apple TV 4K", // Apple TV 4K
+};
+
+export function formatDeviceInfo(deviceInfo: any): string | undefined {
```

Comment:
I think it is better to use the formatted device info from the server. I suggest we add `IdentityClaimBiometricFormattedDeviceInfo` whose value is a string. Then we can avoid duplicate the device info formatting logic.

Reply:
Since the formatted device info can be derived from device info anytime, so I didn't store `IdentityClaimBiometricFormattedDeviceInfo` in db. Updated, see if it looks good?